holoviews/plotting/bokeh/callbacks.py

```diff
@@ -4,115 +4,143 @@
-                    mapping between the name under which the variable
-                    is made available to Python and the specification
-                    of the attribute. The specification should start
-                    with the variable name that is to be accessed and
-                    the location of the attribute separated by
-                    periods.  All models defined by the models and
-                    extra_models attributes can be addressed in this
-                    way, e.g. to get the start of the x_range as 'x'
-                    you can supply {'x': 'x_range.attributes.start'}.
-                    Additionally certain handles additionally make the
-                    cb_data and cb_obj variables available containing
-                    additional information about the event.
+    def _filter_msg(self, msg, ids):
+        """
+        Filter event values that do not originate from the plotting
+        handles associated with a particular stream using their
+        ids to match them.
+        """
+        filtered_msg = {}
+        for k, v in msg.items():
+            if isinstance(v, dict) and 'id' in v:
+                if v['id'] in ids:
+                    filtered_msg[k] = v['value']
+            else:
+                filtered_msg[k] = v
+        return filtered_msg
 
-    * skip        : Conditions when the Callback should be skipped
-                    specified as a list of valid JS expressions, which
-                    can reference models requested by the callback,
-                    e.g. ['pan.attributes.active'] would skip the
-                    callback if the pan tool is active.
 
-    * code        : Defines any additional JS code to be executed,
-                    which can modify the data object that is sent to
-                    the backend.
+    def on_msg(self, msg):
+        for stream in self.streams:
+            handle_ids = self.handle_ids[stream]
+            ids = list(handle_ids.values())
+            filtered_msg = self._filter_msg(msg, ids)
+            processed_msg = self._process_msg(filtered_msg)
+            if not processed_msg:
+                continue
+            stream.update(trigger=False, **processed_msg)
+            stream._metadata = {h: {'id': hid, 'events': self.on_events}
+                                for h, hid in handle_ids.items()}
+        Stream.trigger(self.streams)
+        for stream in self.streams:
+            stream._metadata = {}
 
-    * events      : If the Callback should listen to bokeh events this
-                    should declare the types of event as a list (optional)
 
-    * change      : If the Callback should listen to model attribute
-                    changes on the defined ``models`` (optional)
+    def _init_plot_handles(self):
+        """
+        Find all requested plotting handles and cache them along
+        with the IDs of the models callbacks will be attached to.
```

Comment:
Either an an extra word here or words missing. Either '... along with the IDs the model callbacks will be attached to' or '...along with the IDs of the model callbacks that they will be attached to'.